New Mac keyboards are loaded via
   $mac   $macvendorlayouts = pc(pc105)+macintosh_vndr/%l[1]%(v[1])
in rules/xorg, but function keys are redefined in macintosh_vndr/us(extended) 
which breaks terminal switching for us layout (see #370564).
Here we move definitions which are performed in pc(pc105) into a new oldmac
variant, which is used only with old Mac keyboards.

Not submitted yet.

Index: symbols/macintosh_vndr/us
===================================================================
--- symbols/macintosh_vndr/us.orig
+++ symbols/macintosh_vndr/us
@@ -67,38 +67,6 @@
     key <SPCE> {	[  space			]	};
     // End alphanumeric section
 
-    // Begin "Function" section
-    key <FK01> {	[  F1		]	};
-    key <FK02> {	[  F2		]	};
-    key <FK03> {	[  F3		]	};
-    key <FK04> {	[  F4		]	};
-    key <FK05> {	[  F5		]	};
-    key <FK06> {	[  F6		]	};
-    key <FK07> {	[  F7		]	};
-    key <FK08> {	[  F8		]	};
-    key <FK09> {	[  F9		]	};
-    key <FK10> {	[  F10		]	};
-    key <FK11> {	[  F11		]	};
-    key <FK12> {	[  F12		]	};
-    key <FK13> {	[  Print	]	};
-    key <FK14> {	[  Scroll_Lock	]	};
-    key <FK15> {	[  Pause	]	};
-    // End "Function" section
-
-    // Begin "Editing" section
-    key  <INS> {	[  Insert	]	};
-    key <HOME> {	[  Home		]	};
-    key <PGUP> {	[  Prior	]	};
-    key <DELE> {	[  Delete	]	};
-    key  <END> {	[  End		]	};
-    key <PGDN> {	[  Next		]	};
-
-    key   <UP> {	[  Up		]	};
-    key <LEFT> {	[  Left		]	};
-    key <DOWN> {	[  Down		]	};
-    key <RGHT> {	[  Right	]	};
-    // End "Editing" section
-
     // Begin "Keypad" section
     key <NMLK> {	[  Num_Lock,	Pointer_EnableKeys	]	};
     key <KPEQ> {	[  equal				]	};
@@ -124,27 +92,6 @@
     key <KPDL> { 	[  KP_Decimal,	KP_Delete		]	};
     // End "Keypad" section
 
-    // Begin "Modifier" section
-    key <LFSH> {	[  Shift_L	]	};
-    key <RTSH> {	[  Shift_R	]	};
-    key <LCTL> {	[  Control_L	]	};
-    key <RCTL> {	[  Control_R	]	};
-    key <LALT> {	[  Alt_L, Meta_L ]	};
-    key <RALT> {	[  Alt_R, Meta_R ]	};
-    key <LWIN> {	[  Super_L	]	};
-    key <RWIN> {	[  Super_R	]	};
-    // End "Modifier" section
-
-    // begin modifier mappings
-    modifier_map Shift  { Shift_L , Shift_R };
-    modifier_map Lock   { Caps_Lock };
-    modifier_map Control{ Control_L, Control_R };
-    modifier_map Mod1   { Alt_L, Alt_R, Meta_L, Meta_R };
-    modifier_map Mod2   { Num_Lock };
-    modifier_map Mod4   { Super_L, Super_R };
-    modifier_map Mod5   { Scroll_Lock };
-    // end modifier mappings
-
     include "compose(rwin)"
     include "macintosh_vndr/apple(extended)"
 
@@ -202,3 +149,63 @@
 //    modifier_map Mod3   { Mode_switch };
 
 };
+
+hidden
+xkb_symbols "oldmac" {
+
+    include "macintosh_vndr/us(extended)"
+
+    // Begin "Function" section
+    key <FK01> {	[  F1		]	};
+    key <FK02> {	[  F2		]	};
+    key <FK03> {	[  F3		]	};
+    key <FK04> {	[  F4		]	};
+    key <FK05> {	[  F5		]	};
+    key <FK06> {	[  F6		]	};
+    key <FK07> {	[  F7		]	};
+    key <FK08> {	[  F8		]	};
+    key <FK09> {	[  F9		]	};
+    key <FK10> {	[  F10		]	};
+    key <FK11> {	[  F11		]	};
+    key <FK12> {	[  F12		]	};
+    key <FK13> {	[  Print	]	};
+    key <FK14> {	[  Scroll_Lock	]	};
+    key <FK15> {	[  Pause	]	};
+    // End "Function" section
+
+    // Begin "Editing" section
+    key  <INS> {	[  Insert	]	};
+    key <HOME> {	[  Home		]	};
+    key <PGUP> {	[  Prior	]	};
+    key <DELE> {	[  Delete	]	};
+    key  <END> {	[  End		]	};
+    key <PGDN> {	[  Next		]	};
+
+    key   <UP> {	[  Up		]	};
+    key <LEFT> {	[  Left		]	};
+    key <DOWN> {	[  Down		]	};
+    key <RGHT> {	[  Right	]	};
+    // End "Editing" section
+
+    // Begin "Modifier" section
+    key <LFSH> {	[  Shift_L	]	};
+    key <RTSH> {	[  Shift_R	]	};
+    key <LCTL> {	[  Control_L	]	};
+    key <RCTL> {	[  Control_R	]	};
+    key <LALT> {	[  Alt_L, Meta_L ]	};
+    key <RALT> {	[  Alt_R, Meta_R ]	};
+    key <LWIN> {	[  Super_L	]	};
+    key <RWIN> {	[  Super_R	]	};
+    // End "Modifier" section
+
+    // begin modifier mappings
+    modifier_map Shift  { Shift_L , Shift_R };
+    modifier_map Lock   { Caps_Lock };
+    modifier_map Control{ Control_L, Control_R };
+    modifier_map Mod1   { Alt_L, Alt_R, Meta_L, Meta_R };
+    modifier_map Mod2   { Num_Lock };
+    modifier_map Mod4   { Super_L, Super_R };
+    modifier_map Mod5   { Scroll_Lock };
+    // end modifier mappings
+
+};
Index: rules/base.ml1_s.part
===================================================================
--- rules/base.ml1_s.part.orig
+++ rules/base.ml1_s.part
@@ -2,9 +2,9 @@
   amiga		*			=	xfree68_vndr/amiga(usa1)+%l[1]%(v[1])
   sun4		*			=	latin+sun_vndr/us(sun4)+%l[1]%(v[1])
   sun5		*			=	latin+sun_vndr/us(sun5)+%l[1]%(v[1])
-  old_macintosh	us			=	macintosh_vndr/us(extended)
-  old_macintosh	$macvendorlayouts	=	macintosh_vndr/us(extended)+macintosh_vndr/%l[1]%(v[1])
-  old_macintosh	*			=	macintosh_vndr/us(extended)+%l[1]%(v[1])
+  old_macintosh	us			=	macintosh_vndr/us(oldmac)
+  old_macintosh	$macvendorlayouts	=	macintosh_vndr/us(oldmac)+macintosh_vndr/%l[1]%(v[1])
+  old_macintosh	*			=	macintosh_vndr/us(oldmac)+%l[1]%(v[1])
  $macs		$macvendorlayouts	=	pc(pc105)+macintosh_vndr/%l[1]%(v[1])
  $pcmodels	*			=	pc(%m)+%l[1]%(v[1])
   *		*			=	pc(pc105)+%l[1]%(v[1])
Index: rules/base.ml_s.part
===================================================================
--- rules/base.ml_s.part.orig
+++ rules/base.ml_s.part
@@ -8,11 +8,11 @@
   sun5		*			=	latin+sun_vndr/us(sun5)+%l%(v)
   pc98		nec_vndr/jp		=	nec_vndr/jp(pc98)
   abnt2		br			=	pc(pc104)+br
-  old_macintosh	us			=	macintosh_vndr/us(extended)
-  old_macintosh	en_US			=	macintosh_vndr/us(extended)
-  old_macintosh	$macvendorlayouts	=	macintosh_vndr/us(extended)+macintosh_vndr/%l%(v)
-  old_macintosh	$nonlatin		=	macintosh_vndr/us(extended)+%l%(v):2
-  old_macintosh	*			=	macintosh_vndr/us(extended)+%l%(v)
+  old_macintosh	us			=	macintosh_vndr/us(oldmac)
+  old_macintosh	en_US			=	macintosh_vndr/us(oldmac)
+  old_macintosh	$macvendorlayouts	=	macintosh_vndr/us(oldmac)+macintosh_vndr/%l%(v)
+  old_macintosh	$nonlatin		=	macintosh_vndr/us(oldmac)+%l%(v):2
+  old_macintosh	*			=	macintosh_vndr/us(oldmac)+%l%(v)
  $macs		en_US			=	pc(pc105)+macintosh_vndr/us(extended)
  $macs		$macvendorlayouts	=	pc(pc105)+macintosh_vndr/%l%(v)
  $pcmodels      intl   			=       us(%mcompose)
